compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -683,11 +683,11 @@ impl<'p, 'tcx> MatchVisitor<'p, 'tcx> {
             let end_span = semi_span.shrink_to_lo();
             let count = witnesses.len();
 
-            let_suggestion = Some(if bindings.is_empty() {
-                SuggestLet::If { start_span, semi_span, count }
-            } else {
-                SuggestLet::Else { end_span, count }
-            });
+            if bindings.is_empty() {
+                let_suggestion = Some(SuggestLet::If { start_span, semi_span, count });
+            } else if scrut.is_some() {
```

Comment:
could you “float out” the `scrut.is_some()`, i.e., move it the closest `if`/`let` further up? this way, you also fix:

```rs
fn main() {
    let Some(());
}
```

```
error[E0005]: refutable pattern in local binding
 --> src/main.rs:2:9
  |
2 |     let Some(());
  |         ^^^^^^^^ pattern `None` not covered
  |
  = note: `let` bindings require an "irrefutable pattern", like a `struct` or an `enum` with only one variant
  = note: for more information, visit https://doc.rust-lang.org/book/ch18-02-refutability.html
  = note: the matched value is of type `Option<()>`
help: you might want to use `if let` to ignore the variant that isn't matched
  |
2 |     if let Some(()); { todo!() }
  |     ++               +++++++++++

```

Reply:
That's exactly what I attempted first, and it caused a bunch of existing tests with no bindings to start failing.  I should've paid more attention, because you're right: the suggestion is just as useless in those cases.